### ISSUE TITLE
Optimizer Bug on 'if' with logical 'and'

### ISSUE THREAD

**Issue by oliverschmidt** [cc65/cc65#735]
The program below prints "failures: 0", as expected, when built without optimizations. But, when built with optimizations, it prints "failures: 1". :-(
```nesC
#include <stdio.h>

unsigned char failures = 0;

int main(void)
{
  int i;

  i = 0;
  if ((i > 1) && (i < 3)) {
    failures++;
  }

  printf("failures: %u\n", failures);
  return failures;
}
```

**Comment by DarwinNE** [comment]
Is it possible that this behavior may be similar to what described in #367?

**Comment by oliverschmidt** [comment]
I can‘t assess that but I don‘t see any indication leading to it.

**Comment by laubzega** [comment]
This is a bug in OptPrecalc, where incorrect assumption on ADC #0/SBC #0 while A equals 0 not affecting any flags is made. V is in fact cleared and in the testcase the very next instruction is a BVS.

Fix forthcoming.

**Comment by laubzega** [comment]
The patch is relatively simple, it just disables the optimization if BVC or BVS is the next instruction. I checked the code generator and optimizer and it seems that the only situation when these instructions are generated is right after SBC, so we don't have to worry about V being tested further down the line.

I first tried to be smart about this optimization by replacing BVC with appropriate "always taken" branch and by removing "BVS" altogether, but these attempts were defeated by the subsequent efforts of the optimizer, producing faulty code. Could be another bug, could be my misunderstanding.
